src/wp-includes/class-wp-query.php

```diff
@@ -3434,6 +3434,9 @@ public function next_post() {
 	 */
 	public function the_post() {
 		global $post;
+		if ( ! $this->in_the_loop ) {
+			update_post_author_caches( $this->posts );
+		}
```

Comment:
Why does this condition need to differ from the _loop started_ check below?

Reply:
So you only call this function at the start of the loop and only call it once.